File: backend/app/schemas/tailor_rule.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class TailorRuleCreate(BaseModel):
    application_id: Optional[str] = None  # None = global rule
    rule_text: str
    is_enabled: bool = True


class TailorRuleUpdate(BaseModel):
    rule_text: Optional[str] = None
    is_enabled: Optional[bool] = None


class TailorRuleRead(BaseModel):
    id: str
    user_id: Optional[str] = None
    application_id: Optional[str] = None
    rule_text: str
    is_enabled: bool
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/types/tailor-rule.ts
export interface TailorRule {
    id: string;
    user_id: string | null;
    application_id: string | null;
    rule_text: string;
    is_enabled: boolean;
    created_at: string;
    updated_at: string;
}

export interface TailorRuleCreate {
    application_id?: string | null;
    rule_text: string;
    is_enabled?: boolean;
}

export interface TailorRuleUpdate {
    rule_text?: string;
    is_enabled?: boolean;
}
